move/move-ibc/sources/state_lens_lc.move

```diff
@@ -1,23 +1,181 @@
+        state_root_offset: u16,
+        /// the offset at which we extract the bytes32 storage root (of the ibc contract on the l2) from the l2 consensus state
+        /// storage_root = consensus_state[storage_root_offset:storage_root_offset+32]
+        storage_root_offset: u16
+    }
+
+    struct ConsensusState has copy, drop, store {
+        /// Timestamp of the execution layer.
+        timestamp: u64,
+        /// State root of the execution layer.
+        state_root: vector<u8>,
+        /// Storage root of the ibc contract extracted from the state root.
+        storage_root: vector<u8>
+    }
+
+    struct Header has copy, drop {
+        l1_height: Height,
+        l2_height: Height,
+        /// Proof of the L2 consensus state as stored in the state of the L1.
+        l2_consensus_state_proof: vector<u8>,
+        l2_consensus_state: vector<u8>
+    }
 
     public fun create_client(
-        _ibc_signer: &signer,
-        _client_id: u32,
-        _client_state_bytes: vector<u8>,
-        _consensus_state_bytes: vector<u8>
+        ibc_signer: &signer,
+        client_id: u32,
+        client_state_bytes: vector<u8>,
+        consensus_state_bytes: vector<u8>
     ): (vector<u8>, vector<u8>) {
-        (vector::empty(), vector::empty())
+        let client_state = decode_client_state(client_state_bytes);
+        let consensus_state = decode_consensus_state(consensus_state_bytes);
+
+        assert!(
+            client_state.l2_latest_height != 0 && consensus_state.timestamp != 0,
+            E_INVALID_CLIENT_STATE
+        );
+
+        let consensus_states = smart_table::new<u64, ConsensusState>();
+        smart_table::upsert(
+            &mut consensus_states,
+            client_state.l2_latest_height,
+            consensus_state
+        );
+
+        let state = State { client_state: client_state, consensus_states: consensus_states };
+
+        let store_constructor =
+            object::create_named_object(ibc_signer, bcs::to_bytes<u32>(&client_id));
+        let client_signer = object::generate_signer(&store_constructor);
+
+        move_to(&client_signer, state);
+
+        (client_state_bytes, consensus_state_bytes)
     }
 
-    public fun latest_height(_client_id: u32): u64 {
-        0
+    public fun latest_height(client_id: u32): u64 acquires State {
+        let state = borrow_global<State>(get_client_address(client_id));
+        state.client_state.l2_latest_height
     }
 
     public fun update_client(
-        _client_id: u32, _client_msg: vector<u8>
-    ): (vector<u8>, vector<vector<u8>>, vector<u64>) {
-        (vector::empty(), vector::empty(), vector::empty())
+        client_id: u32, client_msg: vector<u8>
+    ): (vector<u8>, vector<vector<u8>>, vector<u64>) acquires State {
+        let header = decode_header(client_msg);
+
+        let state = borrow_global_mut<State>(get_client_address(client_id));
+
+        assert!(
+            cometbls_lc::verify_membership(
```

Comment:
perhaps we should pull the ics23 verification into a different library

Reply:
its already in `ics23` library but here, we actually wanna call the light client bc it will read it's own state to do the verification against.